src/cli/main.py

```diff
@@ -0,0 +1,291 @@
+    resource = "https://management.azure.com/"
+    data = {
+        "grant_type": "client_credentials",
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "resource": resource,
+    }
+    resp = requests.post(auth_url, data=data)
+    try:
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError:
+        print(resp.json())
+        raise
+
+    resp_json = resp.json()
+    token: str = resp_json["access_token"]
+
+    return token
+
+
+@click.group()
+@click.option("--resource-group", "-g", envvar="AZURE_RESOURCE_GROUP", required=True, help="Azure Resource Group name.")
+@click.option("--client-id", envvar="AZURE_CLIENT_ID", required=True, help="Azure client ID used for authentication.")
+@click.option(
+    "--client-secret", envvar="AZURE_CLIENT_SECRET", required=True, help="Azure client secret used for authentication."
+)
+@click.option("--tenant-id", envvar="AZURE_TENANT_ID", required=True, help="Azure tenant ID.")
+@click.option("--subscription-id", envvar="AZURE_SUBSCRIPTION_ID", required=True, help="Azure subscription ID.")
+@click.pass_context
+def cli(
+    ctx: click.Context, resource_group: str, client_id: str, client_secret: str, subscription_id: str, tenant_id: str
+) -> None:
+    """
+    CLI client for the Azure Image Testing for Linux API
+    """
+    token = auth(tenant_id, subscription_id, client_id, client_secret)
+    ctx.obj = dict()
+    session = requests.session()
+    session.headers.update({"Authorization": f"Bearer {token}"})
+    ctx.obj["session"] = session
+
+    ctx.obj["resource_group"] = resource_group
+    ctx.obj["subscription_id"] = subscription_id
+
+
+@cli.command()
+@click.pass_context
+def list_templates(ctx: click.Context) -> None:
+    """
+    List all the test job templates.
+    """
+    endpoint = _get_endpoint("jobTemplates", ctx.obj["resource_group"], ctx.obj["subscription_id"])
+
+    session = ctx.obj["session"]
+
+    resp = session.get(endpoint)
+    _output_result(resp)
+
+
+@cli.command()
+@click.pass_context
+@click.option("--name", "-n", help="Job template name.", required=True)
+def get_template(ctx: click.Context, name: str) -> None:
+    """
+    Get a test job template.
+    """
+    endpoint = _get_endpoint(f"jobTemplates/{name}", ctx.obj["resource_group"], ctx.obj["subscription_id"])
+
+    session = ctx.obj["session"]
+
+    resp = session.get(endpoint)
+    _output_result(resp)
+
+
+@cli.command()
+@click.pass_context
+@click.option("--name", "-n", help="Job template name.", required=True)
+@click.option("--vm-size", "-s", help="VM size.")
+@click.option("--priority", "-p", help="A comma separated list of LISA test priority to run.")
+@click.option("--case-name", "-c", help="A comma separated list of LISA test case to run.")
+@click.option(
+    "--location",
+    "-l",
+    default="westus3",
+    help="Job template location, it's not required to be changed.",
+)
+@click.option(
+    "--region",
```

Comment:
In the interest of consistency with `--priorities` and `--case-names`, should this parameter also take a comma-separated list?